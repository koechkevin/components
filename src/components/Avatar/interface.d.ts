import { AvatarProps } from 'antd/lib/avatar';
import { CSSProperties, ReactNode } from 'react';

enum ChatStatus {
  online = 'online',
  offline = 'offline',
}

export interface Props extends AvatarProps {
  src?: string;
  dot?: boolean;
  status?: ChatStatus | string;
  size?: number | 'xsmall' | 'small' | 'medium' | 'large' | 'default';
  color?: string;
}
