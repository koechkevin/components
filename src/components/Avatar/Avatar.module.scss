@import '../../styles/global';

.avatar {
  &:global(.ant-avatar-square) {
    border-radius: 8px;
  }

  &:global(.ant-avatar img) {
    background-color: $color-basic-light-gray;
  }
}
