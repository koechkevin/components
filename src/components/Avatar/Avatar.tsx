import { Avatar as AntdAvatar } from 'antd';
import React, { FC } from 'react';

import styles from './Avatar.module.scss';

import { Props } from './interface.d';

export const Avatar: FC<Props> = (props) => {
  const { children, className, icon, src, shape, size, style, color } = props;
  const avatarSizes = {
    small: 32,
    medium: 40,
    large: 80,
    default: 'default',
  };
  const avatarSize = size && (avatarSizes[size] || size);
  const placeholderIcon = src ? icon : 'user';
  const classNames: string = [styles.avatar, className ? className : ''].join(' ');
  const avatarColor = color && !src ? { backgroundColor: color } : {};

  return (
    <AntdAvatar
      src={src}
      shape={shape}
      size={avatarSize}
      className={classNames}
      style={{ minWidth: avatarSize, ...avatarColor, ...style }}
      icon={!children ? placeholderIcon : ''}
    >
      {children}
    </AntdAvatar>
  );
};

Avatar.defaultProps = {
  size: 'default',
};

export default Avatar;
