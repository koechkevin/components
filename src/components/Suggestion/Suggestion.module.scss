@import '../../styles/global';

.suggestion {
  padding: 8px 12px;
  flex-wrap: nowrap;
  cursor: pointer;
  min-width: 360px;

  @include respond-to('xs-l') {
    min-width: 250px;
  }

  .avatar {
    margin-right: 8px;
    display: inline-block;
    position: relative;
  }

  .name {
    color: $color-base-dark;
  }

  .jobPosition {
    color: $color-base-dark 60%;
  }

  .main {
    flex: 1;
    font-size: 13px;
    color: #a0a0a0;
    line-height: 20px;

    .status {
      margin: 0 4px 0 6px;
    }
  }

  .inCall {
    display: flex;
    align-items: center;
    font-size: 12px;
    color: $color-base-dark-gray;

    svg {
      margin-right: 6px;
      font-size: 13px;
    }
  }

  .active {
    color: $color-white;
  }

  .onlineOffline {
    height: 11px;
    width: 11px;
  }
}
