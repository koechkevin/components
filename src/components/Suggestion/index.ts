export { default as Suggestion } from './Suggestion';