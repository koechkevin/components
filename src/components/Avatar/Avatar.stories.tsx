import { number, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import { Col, Row, Typography } from 'antd';
import React from 'react';

import { Center } from '../Center';
import { ListItem } from '../SideBar/CompanyModal';
import Avatar from './Avatar';

import logo from '../../icons/logo.svg';
import mockLogo from '../../icons/mocks/company.png';

const src = 'https://s3.amazonaws.com/uifaces/faces/twitter/bowbrick/128.jpg';
const size = number('size', 200);
const initials = text('initials', 'AA');
const { Text } = Typography;
const companyList: ListItem[] = [
  { companyId: '123', avatarUrl: logo, name: 'Airbus', count: 4, avatarColor: '#fb6d77' },
  { companyId: '1234', avatarUrl: mockLogo, name: 'Pepsi', count: 21, avatarColor: '#46a094' },
  { companyId: '12345', avatarUrl: '', name: 'Airbus', avatarColor: '#5d85ce' },
];

storiesOf('Avatar', module)
  .addDecorator((story) => <Center>{story()}</Center>)
  .add('default', () => <Avatar />, { notes: 'Avatar default styles' })
  .add(
    'with circle shape and image',
    () => (
      <Row>
        <Row gutter={36}>
          <Col span={8}>
            <Text strong>24</Text> xsmall{' '}
          </Col>
          <Col span={8}>
            <Text strong>40</Text> small{' '}
          </Col>
          <Col span={8}>
            <Text strong>80</Text> medium{' '}
          </Col>
        </Row>
        <Row gutter={36}>
          <Col span={8}>
            <Avatar src={src} size="small" />
          </Col>
          <Col span={8}>
            <Avatar src={src} size="medium" />
          </Col>
          <Col span={8}>
            <Avatar src={src} size="large" />
          </Col>
        </Row>
      </Row>
    ),
    {
      notes: 'Avatar with Circle shape and image',
    },
  )
  .add(
    'with square shape and image',
    () => (
      <Row>
        <Row gutter={36}>
          <Col span={8}>
            <Text strong>24</Text> xsmall
          </Col>
          <Col span={8}>
            <Text strong>40</Text> small
          </Col>
          <Col span={8}>
            <Text strong>80</Text> medium
          </Col>
        </Row>
        <Row gutter={36}>
          <Col span={8}>
            <Avatar src={src} shape="square" size="small" />
          </Col>
          <Col span={8}>
            <Avatar src={src} shape="square" size="medium" />
          </Col>
          <Col span={8}>
            <Avatar src={src} shape="square" size="large" />
          </Col>
        </Row>
      </Row>
    ),
    {
      notes: 'Avatar with square shape and image',
    },
  )
  .add(
    'with square shape and initials',
    () => (
      <Row>
        <Row>
          <Col span={12}>
            <Text strong>32</Text> small
          </Col>
          <Col span={12}>
            <Text strong>40</Text> large
          </Col>
        </Row>
        <Row gutter={36}>
          <Col span={12}>
            <Avatar size="small" shape="square" color="#46a094" style={{ fontWeight: 'bold', fontSize: 10 }}>
              {initials}
            </Avatar>
          </Col>
          <Col span={12}>
            <Avatar size="medium" shape="square" color="#8260da" style={{ fontWeight: 'bold', fontSize: 20 }}>
              {initials}
            </Avatar>
          </Col>
        </Row>
      </Row>
    ),
    {
      notes: 'Avatar with Square shape and initials',
    },
  )
  .add(
    'with circle shape and initials',
    () => (
      <Row>
        <Row gutter={36}>
          <Col span={8}>
            <Text strong>32</Text> small
          </Col>
          <Col span={8}>
            <Text strong>40</Text> medium
          </Col>
          <Col span={8}>
            <Text strong>80</Text> large
          </Col>
        </Row>
        <Row gutter={36}>
          <Col span={8}>
            <Avatar size="small" shape="circle" color="#46a094" style={{ fontWeight: 'bold', fontSize: 10 }}>
              {initials}
            </Avatar>
          </Col>
          <Col span={8}>
            <Avatar size="medium" shape="circle" color="#8260da" style={{ fontWeight: 'bold', fontSize: 20 }}>
              {initials}
            </Avatar>
          </Col>
          <Col span={8}>
            <Avatar size="large" shape="circle" color="#8260da" style={{ fontWeight: 'bold', fontSize: 20 }}>
              {initials}
            </Avatar>
          </Col>
        </Row>
      </Row>
    ),
    {
      notes: 'Avatar with Square shape and initials',
    },
  )
  .add('with custom size', () => <Avatar src={src} size={size} />, {
    notes: 'Avatar with custom size',
  })
  .add('with an icon', () => <Avatar icon="user" />, {
    notes: 'Avatar with an icon',
  })
  .add('company logo', () => (
    <Row gutter={32}>
      {companyList.map((company) => (
        <Col span={8} key={company.companyId}>
          <Avatar src={company.avatarUrl} size="medium" shape="square" color={company.avatarColor}>
            {company.name[0]}
          </Avatar>
        </Col>
      ))}
    </Row>
  ));
