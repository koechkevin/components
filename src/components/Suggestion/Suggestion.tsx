import { faPhone } from '@fortawesome/pro-light-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { Row } from 'antd';
import React, { FC, MouseEvent } from 'react';
import { useMedia } from 'react-use';
import { SuggestionsUser } from '../../typings/Message';

import { getNameInitials, ProductType } from '../../utils';
import Avatar from '../Avatar/Avatar';
import styles from './Suggestion.module.scss';

interface Props extends SuggestionsUser {
  isActive: boolean;
  onClick: (e: MouseEvent) => void;
  onMouseOver: (e: MouseEvent) => void;
}

const Suggestion: FC<Props> = (props) => {
  const {
    name,
    icon,
    avatar,
    isActive,
    status,
    inCall,
    onClick,
    productId,
    signature,
    iconColor,
    onMouseOver,
    avatarColor,
  } = props;
  const isCandidate: boolean = productId !== ProductType.Company;
  const avatarStyles = avatar ? {} : { backgroundColor: avatarColor };
  const isMobile = useMedia('(max-width: 575px)');

  return (
    <Row
      type="flex"
      align="middle"
      onClick={onClick}
      onMouseOver={onMouseOver}
      className={styles.suggestion}
      style={{ background: isActive ? '#0050c8' : '' }}
    >
      <div className={styles.avatar}>
        <Avatar
          size="xsmall"
          src={avatar}
          status={status}
          style={avatarStyles}
          shape={isCandidate ? 'square' : 'circle'}
        >
          {getNameInitials(name)}
        </Avatar>
      </div>
      <span className={`${styles.name} ${isActive ? styles.active : ''}`}>{name}</span>
      <div className={styles.main}>
        <span className={styles.status}>
          {icon && (
            <FontAwesomeIcon
              icon={icon}
              className={styles.onlineOffline}
              style={{
                color: iconColor,
              }}
            />
          )}
        </span>
        {!isMobile && <span className={`${styles.jobPosition} ${isActive ? styles.active : ''}`}>{signature}</span>}
      </div>
      {inCall && (
        <div className={styles.inCall}>
          <FontAwesomeIcon icon={faPhone} />
          <span>In Call</span>
        </div>
      )}
    </Row>
  );
};

export default Suggestion;
