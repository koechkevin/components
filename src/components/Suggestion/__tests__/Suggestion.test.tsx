import { faCircle } from '@fortawesome/pro-light-svg-icons';
import { mount } from 'enzyme';
import React from 'react';

import { Avatar } from '../../Avatar/Avatar';
import Suggestion from '../Suggestion';

describe('Tests for Typing component', () => {
  const props = {
    id: '1',
    name: 'Test User',
    productId: 'candidate-portal',
    chatStatus: 'online',
    avatarColor: 'red',
    signature: 'CEO',
    label: 'label',
    status: '',
    inCall: true,
    icon: faCircle,
    iconColor: 'red',
    isActive: true,
    onClick: jest.fn(),
    onMouseOver: jest.fn(),
  };

  const wrapper = mount(<Suggestion {...props} />);

  it('should render Typing without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correct avatar in suggestions', () => {
    expect(
      wrapper
        .find(Avatar)
        .first()
        .props().shape,
    ).toEqual('square');
  });

  it('should contain circle avatar for company portal user', () => {
    const companyUserProps = {
      ...props,
      productId: 'company-portal',
    };

    const updatedWrapper = mount(<Suggestion {...companyUserProps} />);
    expect(
      updatedWrapper
        .find(Avatar)
        .first()
        .props().shape,
    ).toEqual('circle');
  });

  it('should render the name of users in the list', () => {
    expect(wrapper.contains('Test User')).toBe(true);
  });

  it('should render the signature of users in the list', () => {
    expect(wrapper.contains('CEO')).toBe(true);
  });

  it('should handle onClick for listed users', () => {
    const suggestion = wrapper.find('.suggestion').first();
    suggestion.simulate('click');

    expect(props.onClick).toHaveBeenCalled();
  });

  it('should render in call when true', () => {
    expect(wrapper.contains('In Call')).toBe(true);
  });

  it('should render not render in call when false', () => {
    const updatedProps = {
      ...props,
      inCall: false,
    };

    const updatedWrapper = mount(<Suggestion {...updatedProps} />);
    expect(updatedWrapper.contains('In Call')).toBe(false);
  });
});
