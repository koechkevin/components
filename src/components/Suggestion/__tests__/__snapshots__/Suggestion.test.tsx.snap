// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Typing component should render Typing without crashing 1`] = `ReactWrapper {}`;
